core/src/main/scala/org/apache/spark/ui/UIUtils.scala

```diff
@@ -354,7 +354,7 @@ private[spark] object UIUtils extends Logging {
         {completed}/{total}
         { if (failed > 0) s"($failed failed)" }
         { if (skipped > 0) s"($skipped skipped)" }
-        { if (killed > 0) s"($killed killed)" }
+        { reasonToNumKilled.map { case (reason, count) => s"($count killed: $reason)" } }
```

Comment:
Would be good to sort it in some order and expose it.
Either by desc count or alphabetically : so that updates do not reorder arbitrarily.